assets/js/script.js

```diff
@@ -4,124 +4,80 @@ $(document).ready(function() {
-    $(this).parent().addClass('active')
+  setState({
+    materials: materials
   })
-
-  el.letterTab1.on('click', function() {
-    el.swatch.removeClass('active')
-    el.swatch.each(function() {
-      if ($(this).data("material-swatch") == letter1Choice) {
-        $(this).addClass("active")
-      }
-    })
+})
+$(document).on('click', '.js-tab', function(evt) {
+  setState({
+    activeLetter: $(this).data('index')
   })
+})
 
-  el.letterTab2.on('click', function() {
-    el.swatch.removeClass('active')
-    el.swatch.each(function() {
-      if ($(this).data("material-swatch") == letter2Choice) {
-        $(this).addClass("active")
-      }
-    })
-  })
+function setState(newState) {
+  state = Object.assign({}, state, newState)
+  render()
+}
 
-  el.input.on("input", function() {
-    var $inputText = $(this).val()
+function updateCustomInfo() {
+  var productDescription = 'Danny-Black'
+  if( state.letters.length > 0 ) {
+    productDescription += ` / ${state.letters[0].toUpperCase()}-${state.materials[0]}`
+  }
+  if( state.letters.length > 1 ) {
+    productDescription += ` / ${state.letters[1].toUpperCase()}-${state.materials[1]}`
+  }
 
-    if ($inputText.length > 0) {
-      $("#bagLetter2").hide()
-      renderLetter(1, $inputText.charAt(0), letter1Choice)
-      if ($inputText.length > 1) {
-        renderLetter(2, $inputText.charAt(1), letter2Choice)
-        $("input").blur()
-      }
-      updateCustomInfo()
-    } else {
-      $("#tab-letter1").html("")
-      $("#tab-letter2").html("")
-      $("#bagLetter1").css({ display: "none" })
-      $("#bagLetter2").css({ display: "none" })
-    }
-  })
+  $("#custombar-custom-info").val(productDescription);
+}
 
-  el.swatch.on('click', function() {
-    el.swatch.removeClass("active")
-    $(this).addClass("active")
-    var text = el.input.val()
+function render() {
+  console.log('rendering')
 
-    if ($("#tab-letter1").hasClass("active")) {
-      letter1Choice = $(this).data("material-swatch")
-      renderLetter(1, text.charAt(0).toUpperCase(), letter1Choice)
-    } else if ($("#tab-letter2").hasClass("active")) {
-      letter2Choice = $(this).data("material-swatch")
-      renderLetter(2, text.charAt(1).toUpperCase(), letter2Choice)
-    } else {
-      console.warn('Unknown letter active')
-    }
+  // reset
+  $('.js-letters').html('')
+  $('.js-tab').html('')
+  $('.js-tab').removeClass('active')
+  $('.js-swatch').removeClass('active')
 
-    updateCustomInfo()
-  })
-  function renderLetter(index, letter, swatch) {
-    var height = $(".letters-wrapper").height()
-    var letterPath = (window.baseUrl || "") + "assets/img/letters/" + letter.toUpperCase() + "-" + swatch + ".png"
-    $("#tab-letter" + index).css({ background: "url(" + letterPath + ") no-repeat", backgroundSize: "contain", backgroundPosition: "center" })
-    $("#bagLetter" + index).attr("src", letterPath)
-    $("#bagLetter" + index).show().css({ height: height })
+  // render
```

Comment:
then, we update the DOM according to the current state. here we're appending some images and adding some class names.

one of the powerful things here is that this is idempotent and declarative. so given the same state object, the UI will always render exactly the same regardless of the previous state.

this way, in the event handlers and other code, we only need to make sure the state object is being updated correctly. the actual DOM manipulation is handled by this render method.